sdks/java/io/elasticsearch/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTest.java

```diff
@@ -0,0 +1,303 @@
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ElasticsearchIOTest.class);
+
+  private static final String DATA_DIRECTORY = "target/elasticsearch";
+  public static final String ES_INDEX = "beam";
+  public static final String ES_TYPE = "test";
+  private static final String ES_IP = "127.0.0.1";
+  private static final long NB_DOCS = 400L;
+
+  private static int esHttpPort;
+
+  private static transient Node node;
+  private static ElasticsearchIO.ConnectionConfiguration connectionConfiguration;
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    ServerSocket serverSocket = new ServerSocket(0);
+    esHttpPort = serverSocket.getLocalPort();
+    serverSocket.close();
+    connectionConfiguration = ElasticsearchIO.ConnectionConfiguration
+        .create(new String[]{"http://" + ES_IP + ":" + esHttpPort}, ES_INDEX, ES_TYPE);
+    FileUtils.deleteDirectory(new File(DATA_DIRECTORY));
+    LOGGER.info("Starting embedded Elasticsearch instance ({})", esHttpPort);
+    Settings.Builder settingsBuilder =
+        Settings.settingsBuilder()
+            .put("cluster.name", "beam")
+            .put("http.enabled", "true")
+            .put("node.data", "true")
+            .put("path.data", DATA_DIRECTORY)
+            .put("path.home", DATA_DIRECTORY)
+            .put("node.name", "beam")
+            .put("network.host", ES_IP)
+            .put("http.port", esHttpPort)
+            .put("index.store.stats_refresh_interval", 0);
+    node = NodeBuilder.nodeBuilder().settings(settingsBuilder).build();
+    LOGGER.info("Elasticsearch node created");
+    node.start();
+  }
+
+  @Before
+  public void before() throws Exception {
+    IndicesAdminClient indices = node.client().admin().indices();
+    IndicesExistsResponse indicesExistsResponse =
+        indices.exists(new IndicesExistsRequest(ES_INDEX)).get();
+    if (indicesExistsResponse.isExists()) {
+      indices.prepareDelete(ES_INDEX).get();
+    }
+  }
+
+  private List<String> createDocuments(long nbDocs, boolean insertErrors){
+    String[] scientists =
+        { "Einstein", "Darwin", "Copernicus", "Pasteur", "Curie", "Faraday", "Newton", "Bohr",
+            "Galilei", "Maxwell" };
+    ArrayList<String> data = new ArrayList<>();
+    for (int i = 0; i < nbDocs; i++) {
+      int index = i % scientists.length;
+      //insert 2 malformed documents is asked to do so
+      if (insertErrors && (i == 6 || i == 7)){
+        data.add(String.format("{\"scientist\";\"%s\", \"id\":%d}", scientists[index], i));
+      } else {
+        data.add(String.format("{\"scientist\":\"%s\", \"id\":%d}", scientists[index], i));
+      }
+    }
+    return data;
+  }
+  private void insertTestDocuments(long nbDocs) throws Exception {
+    Client client = node.client();
+    final BulkRequestBuilder bulkRequestBuilder = client.prepareBulk().setRefresh(true);
+    List<String> data = createDocuments(nbDocs, false);
+    for (String document:data){
```

Comment:
Whitespace in this line and in general in this file is also inconsistent - please reformat it as well.